packages/web-frontend/src/containers/EnlargedDialog/EnlargedDialog.js

```diff
@@ -52,39 +46,89 @@ const STATUS = {
   ERROR: 'error',
 };
 
-const EnlargedDialogComponent = ({
-  isDrillDownVisible,
-  onCloseOverlay,
-  viewContent,
-  errorMessage,
-  organisationUnitName,
-  onDrillDown,
-  onSetDateRange,
-  isLoading,
-  isVisible,
-  startDate,
-  endDate,
-  projectCode,
-}) => {
+const EnlargedDialogComponent = props => {
+  const {
+    onCloseOverlay,
+    contentByLevel,
+    errorMessage,
+    organisationUnitName,
+    onSetDateRange,
+    startDate,
+    endDate,
+    isLoading,
+    initialViewContent,
+    projectCode,
+    infoViewKey,
+    fetchViewData,
+  } = props;
   const exportRef = React.useRef(null);
   const [exportDialogIsOpen, setExportDialogIsOpen] = React.useState(false);
   const [isExporting, setIsExporting] = React.useState(false);
+
   const [exportStatus, setExportStatus] = React.useState(STATUS.IDLE);
+  const [drillDownState, setDrillDownState] = React.useState({
+    drillDownLevel: 0,
+    parameterLink: null,
+    parameterValue: null,
+  });
+
+  const viewContent = contentByLevel?.[drillDownState.drillDownLevel]?.viewContent;
+
+  const { organisationUnitCode, dashboardGroupId, viewId } = getInfoFromInfoViewKey(infoViewKey);
 
-  if (!isVisible) {
-    return null;
-  }
+  React.useEffect(() => {
+    if (!viewId) return;
+
+    // No need to refetch if everything is default
+    if (!startDate && !endDate && !drillDownState.drillDownLevel === 0) return;
+
+    const options = {
+      viewContent: {
+        infoViewKey,
+        organisationUnitCode,
+        viewId,
+        dashboardGroupId,
+      },
+      projectCode,
+      startDate,
+      endDate,
+      isExpanded: true,
+      ...drillDownState,
+    };
+    fetchViewData(options);
+    // TODO: Only refetching on parameterValue change is a bit hacky... We rely on not refetching here...
+  }, [startDate, endDate, drillDownState.parameterValue, infoViewKey]);
```

Comment:
This is the section I seem to be having issues with.

I want it to reload on open if and only if:
 - The expanded date range is different to the date range for the small one.
 - The report is a matrix (in small form, it just showed a placeholder)

(https://github.com/beyondessential/tupaia/pull/1623 may help with not reloading after opening)

After opening, it should reload only on
 - Date range change
 - Drill down

Note:
 - It should NOT reload on drill up (this relates to the TODO)

Also, should we consider the cases that we don't support, such as the infoViewKey changing while the report is open?

Reply:
Discussed with Tom Caiger.